LayoutTests/resources/ipc.js

```diff
@@ -2,10 +2,31 @@ function randomIPCID() {
     return Math.floor(Math.random() * 10000) + 1;
 }
 
+function connectionSendWithPromisedReply(connection, destinationID, messageName, args) {
+    return new Promise((resolve, reject) => {
+        function cb(result) {
+            if (typeof result == "object" && typeof result.arguments === "object")
+                resolve(result);
+            else
+                reject();
+        }
+        connection.sendWithAsyncReply(destinationID, messageName, args, cb);
+    });
+}
+
+if (window.IPC) {
+    IPC.sendWithPromisedReply = function(processTarget, destinationID, messageName, args) {
+        return connectionSendWithPromisedReply(IPC.getConnectionForProcessTarget(processTarget), destinationID, messageName, args);
+    }
+}
+
 function asyncFlush(processTarget) {
```

Comment:
Can we make this function async and use `await` in the function instead of manually calling then?